### ISSUE TITLE
Missing weapon sprites or glitchy Totk

### ISSUE THREAD

**Issue by WimpyWat3R** [yuzu-emu/yuzu#10327]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Mainline 1435

### Description of Issue

I have only seen this problem with the weapons the bows, shields, materials, food, everything else is perfect but the weapon sprites are glitchy and sometimes missing.
![unnamed](https://github.com/yuzu-emu/yuzu/assets/133458835/471209a7-5a43-418a-9b14-99b0edc689eb)

### Expected Behavior

Should not have glitchy weapon sprites

### Reproduction Steps

Default settings, Only occurs when on AMD my Nvidia friends have not run into this issue even if i replicate their settings, 

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11483024/yuzu_log.txt)

[   0.147792] Input <Info> input_common\drivers\udp_client.cpp:UDPClient:139: Udp Initialization started
[   0.147807] Input <Info> input_common\drivers\udp_client.cpp:StartCommunication:331: Starting communication with UDP input server on 127.0.0.1:26760
[   0.288108] Input <Info> input_common\drivers\joycon.cpp:Joycons:22: Joycon driver Initialization started
[   0.387333] Frontend <Info> yuzu\main.cpp:GMainWindow:360: yuzu Version: yuzu 1431
[   0.387354] Frontend <Info> yuzu\main.cpp:LogRuntimes:239: MSVC Compiler: 1934 Runtime: 14.34.31938.0
[   0.387355] Frontend <Info> yuzu\main.cpp:LogRuntimes:248: Qt Compile: 5.15.2 Runtime: 5.15.2
[   0.387356] Frontend <Info> yuzu\main.cpp:GMainWindow:376: Host CPU: AMD Ryzen 5 5500                                | AVX2 | FMA
[   0.387360] Frontend <Info> yuzu\main.cpp:GMainWindow:378: Host CPU Cores: 6
[   0.387361] Frontend <Info> yuzu\main.cpp:GMainWindow:381: Host CPU Threads: 12
[   0.387478] Frontend <Info> yuzu\main.cpp:GMainWindow:382: Host OS: Windows 11 Version 22H2 (Build 22621.1702)
[   0.387490] Frontend <Info> yuzu\main.cpp:GMainWindow:384: Host RAM: 15.87 GiB
[   0.387491] Frontend <Info> yuzu\main.cpp:GMainWindow:385: Host Swap: 17.00 GiB
[   0.387542] Frontend <Info> yuzu\main.cpp:GMainWindow:390: Host Timer Resolution: 0.5001 ms
[   1.169186] Common.Filesystem <Error> common\fs\fs.cpp:IterateDirEntries:415: Filesystem object at path=G:/Games/Games/Emulators/Switch Games does not exist
[   1.169192] Common.Filesystem <Error> common\fs\fs.cpp:IterateDirEntries:415: Filesystem object at path=G:/Games/Games/Emulators/Switch Games does not exist

### System Configuration

CPU: Ryzen 5 5500
GPU/Driver: Rx 66500 xt
RAM: G.skill Ripjaws V 2x8
OS: Windows 11

**Comment by sharkiller** [comment]
yes. this happen a lot to weapons with attachments on all version of yuzu. some are missing the attachment, some are missing the base weapon, and some are missing completly.

**Comment by simper** [comment]
I have the exact same issue on AMD RX6600 with Vulkan. I tried both mainline build and EA build 3591, same result.

**Comment by javierjsp** [comment]
+1 Here

**Comment by chiptunesoprano** [comment]
Looks like yuzu caches current inventory icons at AppData\Roaming\yuzu\nand\user\save\cache\0000000000000000\icon

I noticed I could temporarily, sometimes, fix this by deleting this folder and letting yuzu recache next time I loaded the game. Well it would fix incorrect icons but not invisible attachments, since they'd be cached invisible. I have nvidia and this was still a thing.

**Comment by lhecker** [comment]
I've never worked on or read the source code for yuzu, but the issue seemingly involves two bugs at once. If I'm guessing right, this is the offending code:
https://github.com/yuzu-emu/yuzu/blob/1e398e6c36bf8839ca530497fee3772f51d62126/src/core/file_sys/savedata_factory.cpp#L203

The first problem is that, unlike regular saves, the cache directory is not keyed by user ID. This results in missing/broken/wrong textures whenever you use more than one profile to play games on a single PC. It would need to look like that instead (or similar):
```cpp
        return fmt::format("{}save/cache/{:016X}{:016X}/{:016X}", out, user_id[1], user_id[0], title_id);
```

But I'm not sure if that's considered a legitimate path anymore. The PR that introduced this feature (#1880) only notes:
> Done at path `/cache/<title_id>`, which is our best guess for how the system actually does it.

Furthermore I find it surprising that the `title_id` is all 0. Shouldn't it be the ID of the game? I unfortunately don't yet understand the logic behind PRs like #10237, so I'm not sure if that's just a temporary hack by yuzu, but I suspect that it would lead to path collisions with other games in the future.

**Comment by Flax4** [comment]
I had this issue with an Nvidia card, so not unique to AMD. I fixed it by disabling "Use asynchronous shader building (Hack)"

**Comment by Doguinhoo** [comment]
Also have this problem, and to me it not only happens to fused weapon icons, it also mismatches save icons, camera photos, hyrule compendium entries, etc. (sometimes fused weapons' icons will be the image of a save file, and hyrule compendium entries will have random photos in them instead of the item photo)

**Comment by AM800Mats** [comment]
I'm also experiencing this issue (with weapons only). Entire weapons are invisible or parts of them are missing.

**Comment by javierjsp** [comment]
as a workaround, I drop the weapon with the sprite problem to the floor and grab again, and now the sprite is good, At Least for me.

AMD GPU here, Rx580

**Comment by bloock-w** [comment]
nvidia gpu, same issue

**Comment by weskoop** [comment]
builds from other profiles also show in auto-build history

**Comment by KrisCris** [comment]
FYI, this also happens to the Autobuild history:

![82F32E3F-BD8E-4C34-953D-F3C5C4666423](https://github.com/yuzu-emu/yuzu/assets/38860226/d156d265-b6be-444f-b974-ab9620fafaf8)

**Comment by Doguinhoo** [comment]
the only thing that fixed it for me was turning off async shader builder, but that adds a lot of stutter.
